core/domain/learner_progress_services.py

```diff
@@ -868,6 +1322,93 @@ def _get_filtered_collection_playlist_summaries(
         nonexistent_playlist_collection_ids)
 
 
+def get_story_summary_dicts(user_id, story_summaries):
+    """Returns a displayable summary dict of the the story summaries
+    given to it.
+
+    Args:
+        user_id: str. The id of the learner.
+        story_summaries: list(StorySummary). A list of the
+            summary domain objects.
+
+    Returns:
+        list(dict). The summary dict objects corresponding to the given summary
+        domain objects.
+    """
+    summary_dicts = []
+    for story_summary in story_summaries:
+        story = story_fetchers.get_story_by_id(story_summary.id)
+        topic = topic_fetchers.get_topic_by_id(story.corresponding_topic_id)
+        summary_dicts.append({
+            'id': story_summary.id,
+            'title': story_summary.title,
+            'node_titles': story_summary.node_titles,
+            'thumbnail_filename': story_summary.thumbnail_filename,
+            'thumbnail_bg_color': story_summary.thumbnail_bg_color,
+            'description': story_summary.description,
+            'url_fragment': story_summary.url_fragment,
+            'story_is_published': (
+                story_services.is_story_published_and_present_in_topic(story)),
+            'completed_node_titles': [
+                node.title for node in (
+                    story_fetchers.get_completed_nodes_in_story(
+                        user_id, story.id))],
+            'all_node_dicts': [
+                node.to_dict() for node in story.story_contents.nodes],
+            'topic_url_fragment': topic.url_fragment,
+            'classroom_url_fragment': (
+                classroom_services.get_classroom_url_fragment_for_topic_id(
+                    story.corresponding_topic_id))
+        })
+
+    return summary_dicts
+
+
+def get_topic_summary_dicts(user_id, topic_summaries):
```

Comment:
Ditto regarding renaming and whether it is called only from the controller.

Reply:
Done.